main.go

```diff
@@ -745,6 +737,36 @@ The validate command expects a configuration file as its only argument. Local fi
 		Description: "versioneer subcommands",
 		Subcommands: versioneer.CliCommands(),
 	},
+	{
+		Name:        "bootentry",
+		Usage:       "bootentry [--select]",
+		Description: "bootentry subcommands",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:    "select",
+				Usage:   "Select the boot entry",
+				Aliases: []string{"s"},
+			},
+		},
+		Before: func(c *cli.Context) error {
+			return checkRoot()
+		},
+		Action: func(c *cli.Context) error {
+			cfg, err := agentConfig.Scan(collector.Directories(constants.GetConfigScanDirs()...), collector.NoLogs)
+			if err != nil {
+				return err
+			}
+			s := c.String("select")
+			// Here we should separate the call between uki and non-uki
+			// uki deals with systemd-boot and non-uki with grub
+			// so its a different path
```

Comment:
should we maybe err out in such case?

Reply:
I plan on expanding this to deal with grub systems as well, it's just not implemented yet :)

Reply:
Sorry, got send before finishing. I don't plan to merge this yet

Reply:
gotcha, no worries